airflow/models/baseoperator.py

```diff
@@ -57,6 +58,9 @@
 from airflow.utils.trigger_rule import TriggerRule
 from airflow.utils.weight_rule import WeightRule
 
+if TYPE_CHECKING:
+    from airflow.utils.task_group import TaskGroup  # pylint: disable=cyclic-import
```

Comment:
Does `TaskGroup` create a cyclic dependency? We have already a few of them and if it's possible to avoid another one we should try to do this

Reply:
There's actually no cyclic dependency if not because of the type checking at the module scope. E.g. in order to add this type hints in baseoperator.py, we need to import `TaskGroup` under `TYPE_CHECKING`
```
task_group: Optional["TaskGroup"] = None,
```

In task_group.py it's the same. In order to put `BaseOperator` in the type hints, `BaseOperator` needs to be imported.